lib/web/desktop_playback.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package web
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/session"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/websocket"
+)
+
+const actionPlayPause = "play/pause"
+
+const (
+	playStatePlaying  = "playing"
+	playStatePaused   = "paused"
+	playStateFinished = "finished"
+)
+
+// playbackAction is a message passed from the playback client
+// to the server over the websocket connection in order to modify
+// the playback state.
+type playbackAction struct {
+	// Action is one of actionPlayPause | TODO: actionMove
```

Comment:
This comment looks a bit off. Move the TODO to another line and the part about actionPlayPause to the const itself?